cpp/oneapi/dal/algo/kmeans/test/badarg.cpp

```diff
@@ -0,0 +1,159 @@
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+
+namespace oneapi::dal::kmeans::test {
+
+namespace te = dal::test::engine;
+
+template <typename Method>
+class kmeans_badarg_test : public te::algo_fixture {
+public:
+    static constexpr std::int64_t row_count = 8;
+    static constexpr std::int64_t column_count = 2;
+    static constexpr std::int64_t element_count = row_count * column_count;
+    static constexpr std::int64_t cluster_count = 2;
+    static constexpr std::int64_t huge_cluster_count = 10;
+    static constexpr std::int64_t bad_cluster_count = 2;
+    static constexpr std::int64_t bad_column_count = 3;
+    static constexpr std::int64_t bad_element_count = cluster_count * bad_column_count;
+
+    auto get_descriptor() const {
+        return kmeans::descriptor<float, Method, kmeans::task::clustering>{};
+    }
+
+    table get_train_data(std::int64_t override_row_count = row_count,
+                         std::int64_t override_column_count = column_count) const {
+        ONEDAL_ASSERT(override_row_count * override_column_count <= element_count);
+        return homogen_table::wrap(train_data_.data(), override_row_count, override_column_count);
+    }
+
+    table get_infer_data(std::int64_t override_row_count = row_count,
+                         std::int64_t override_column_count = column_count) const {
+        ONEDAL_ASSERT(override_row_count * override_column_count <= element_count);
+        return homogen_table::wrap(infer_data_.data(), override_row_count, override_column_count);
+    }
+
+    table get_initial_centroids(std::int64_t override_row_count = cluster_count,
+                                std::int64_t override_column_count = column_count) const {
+        ONEDAL_ASSERT(override_row_count * override_column_count <= element_count);
+        return homogen_table::wrap(initial_centroids.data(),
+                                   override_row_count,
+                                   override_column_count);
+    }
+
+    table get_bad_initial_centroids(std::int64_t override_row_count = cluster_count,
+                                    std::int64_t override_column_count = bad_column_count) const {
+        ONEDAL_ASSERT(override_row_count * override_column_count <= element_count);
+        return homogen_table::wrap(bad_initial_centroids.data(),
+                                   override_row_count,
+                                   override_column_count);
+    }
+
+private:
+    static constexpr std::array<float, element_count> train_data_ = {
+        1.0, 1.0, 2.0, 2.0, 1.0, 2.0, 2.0, 1.0, -1.0, -1.0, -1.0, -2.0, -2.0, -1.0, -2.0, -2.0
+    };
+
+    static constexpr std::array<float, bad_element_count> infer_data_ = {
+        1.0, 1.0, 2.0, 2.0, 1.0, 2.0, 2.0, 1.0, -1.0, -1.0, -1.0, -2.0, -2.0, -1.0, -2.0, -2.0
+    };
+
+    static constexpr std::array<float, bad_element_count> bad_initial_centroids = { 0.0, 0.0, 0.0,
```

Comment:
Possibly should be converted to `dataframe` class

Reply:
Dataframe isn't for manually prepared data, right?